daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala

```diff
@@ -154,8 +156,15 @@ object Ref {
     def assertFromSegments(s: Iterable[String]): DottedName =
       assertRight(fromSegments(s))
 
-    def fromNames(names: ImmArray[Name]): Either[String, DottedName] =
-      Either.cond(names.nonEmpty, new DottedName(names), "No segments provided")
+    def fromNames(names: ImmArray[Name]): Either[String, DottedName] = {
+      val length = names.foldLeft(-1)(_ + _.length + 1)
+      if (length < 0)
+        Left("No segments provided")
```

Comment:
```suggestion
      if (names. empty) {
        Left("No segments provided")
       } else {
      val length = names.view.map(_.length).sum + names.length - 1
```
seemed a bit easier to understand but feel free to ignore

Reply:
caa2888f26c632b9c89f5f3d10cbae9c19b31444